authsignal/client.py

```diff
@@ -0,0 +1,256 @@
+        """Links additional identifiers for the user
+        Args:
+            user_id:  A user's id. This id should be the same as the user_id used in event calls.
+            user_payload:  A dictionary with the key/value of the identifier you want to link {'email': 'test@test.com'}
+        """
+        _assert_non_empty_unicode(user_id, 'user_id')
+        _assert_non_empty_dict(user_payload, 'user_payload')
+
+        headers = {'Content-type': 'application/json',
+                    'Accept': '*/*',
+                    'User-Agent': self._user_agent()}
+        if path is None:
+            path = self._post_identify_url(user_id)
+        params = {}
+        timeout = self.timeout
+
+        try:
+            response = self.session.post(
+                path,
+                auth=requests.auth.HTTPBasicAuth(self.api_key, ''),
+                data=json.dumps(user_payload),
+                headers=headers,
+                timeout=timeout,
+                params=params)
+            if response.status_code > 299:
+                raise ApiException("Identify Failed", path, http_status_code=response.status_code, api_error_message=response.json()["message"])
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), path)
+    
+    def enrol_authenticator(self, user_id, authenticator_payload,  path=None):
+        """Enrols an authenticator like a phone number for SMS on behalf of the user
+        Args:
+            user_id:  A user's id. This id should be the same as the user_id used in event calls.
+            authenticator_payload:  A dictionary with the key/value of the authenticator you want to link {'oobChannel': 'SMS', 'phoneNumber': '+112345677777'}
+        """
+        _assert_non_empty_unicode(user_id, 'user_id')
+        _assert_non_empty_dict(authenticator_payload, 'authenticator_payload')
+
+        headers = {'Content-type': 'application/json',
+                    'Accept': '*/*',
+                    'User-Agent': self._user_agent()}
+        if path is None:
+            path = self._post_enorlment_url(user_id)
+        params = {}
+        timeout = self.timeout
+
+        try:
+            response = self.session.post(
+                path,
+                auth=requests.auth.HTTPBasicAuth(self.api_key, ''),
+                data=json.dumps(authenticator_payload),
+                headers=headers,
+                timeout=timeout,
+                params=params)
+            if response.status_code > 299:
+                raise ApiException("Enrol Authenticator Failed", path, http_status_code=response.status_code, api_error_message=response.json()["message"])
+            return response.json()
+        except requests.exceptions.RequestException as e:
+            raise ApiException(str(e), path)
+
+    def _user_agent(self):
+        return 'Authsignal Python v%s' % (self.version)
```

Comment:
f-strings are preferred these days:

```suggestion
        return f'Authsignal Python v{self.version}'```